### PR TITLE
Persist specific error properties in core transaction metadata

### PR DESCRIPTION
**Pull request by vinistevam**
## Explanation

This PR aims to persist specific error properties instead of the whole error object.
It is part of the Transaction Controller alignment effort, and as the extension persists specific fields from `Error` object are into the transaction metadata.
Included specific properties to support both clients.

Note: 
When testing this in the extension frontend, the error data is lost during the API RPC layer and becomes an empty object.

### `@metamask/transaction-controller`

- **ADDED**: create `TxError` type with specific properties to support both clients.
  - Updated `failTransaction` to use the new type.

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/transaction-controller/src/types.ts [hunk 1]
```diff
@@ -689,3 +692,38 @@ export type DefaultGasEstimates = {
    */
   maxPriorityFeePerGas?: string;
 };
+
+/**
+ * Represents an error encountered during a transaction.
+ */
+export type TxError = {
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/types.ts:699]: Should we call this `TransactionError` to be consistent with the other `TransactionX` types?

File: packages/transaction-controller/src/types.ts [hunk 2]
```diff
@@ -689,3 +692,38 @@ export type DefaultGasEstimates = {
...
+
+  /**
+   * The rpc property holds additional information related to the error.
+   */
+  rpc?: unknown;
+
+  /**
+   * The value property holds additional information related to the error.
+   */
+  value?: unknown;
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/types.ts:728]: Should this be removed as it's not being used in `normalizeTxError`?

File: packages/transaction-controller/src/types.ts [hunk 3]
```diff
@@ -689,3 +692,38 @@ export type DefaultGasEstimates = {
    */
   maxPriorityFeePerGas?: string;
 };
+
+/**
+ * Represents an error encountered during a transaction.
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/types.ts:697]:
```suggestion
 * Data concerning an error while processing a transaction.
```

File: packages/transaction-controller/src/TransactionController.ts [hunk 4]
```diff
@@ -208,10 +210,10 @@ export class TransactionController extends BaseController<
 
   private readonly pendingTransactionTracker: PendingTransactionTracker;
 
-  private failTransaction(transactionMeta: TransactionMeta, error: Error) {
+  private failTransaction(transactionMeta: TransactionMeta, error: TxError) {
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:213]: As we can fail a transaction for any error, and only normalize here, should this type remain as `Error` for clarity?

File: packages/transaction-controller/src/types.ts (hunk 2 above)
Comment (vinistevam) [reply packages/transaction-controller/src/types.ts:728]: yeap, I had to accept an `Error`in `normalizedTxError` with that I  removed the `value`  property in `TransactionError`.

File: packages/transaction-controller/src/utils/utils.ts [hunk 5]
```diff
@@ -167,3 +167,19 @@ export function validateIfTransactionUnapproved(
...
   }
 }
+
+/**
+ * Normalizes properties on transaction params.
+ *
+ * @param error - The error to be normalize.
+ * @returns Normalized transaction error.
+ */
+export function normalizeTxError(error: TxError): TxError {
```
[the remark this answers is not in the document]
Comment (vinistevam) [reply packages/transaction-controller/src/utils/utils.ts:177]: Perfect, that makes sense. I had to do it to support your first suggestion.

File: packages/transaction-controller/src/types.ts (hunk 2 above)
Comment (matthewwalsh0) [reply packages/transaction-controller/src/types.ts:766]: Is this still pending removal?

Comment (vinistevam) [reply packages/transaction-controller/src/types.ts]: Apologise, I missed this one. Removed https://github.com/MetaMask/core/pull/1915/commits/c15adfbe0c05beb4a4d2a60cde0ca05853db2b2a

**Review by matthewwalsh0 — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**